modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -1122,6 +1122,32 @@ static IE::PreProcessInfo configurePreProcInfo(const IE::InputInfo::CPtr& ii,
     return info;
 }
 
+using namespace cv::gapi::ie::detail;
+static void configureOutputPrecision(const IE::OutputsDataMap             &outputs_info,
+                                     const ParamDesc::precision_variant_t &output_precision) {
+    switch (output_precision.index()) {
+        case ParamDesc::precision_variant_t::index_of<ParamDesc::precision_t>(): {
+            auto precision    = toIE(cv::util::get<ParamDesc::precision_t>(output_precision));
+            for (auto it : outputs_info) {
+                it.second->setPrecision(precision);
+            }
+            break;
+        }
+        case ParamDesc::precision_variant_t::index_of<ParamDesc::precision_map_t>(): {
+            const auto& precision_map =
+                cv::util::get<ParamDesc::precision_map_t>(output_precision);
+            for (auto it : precision_map) {
+                outputs_info.at(it.first)->setPrecision(toIE(it.second));
```

Comment:
Would be great to show more descriptive array if user specified incorrect layer name

Reply:
Probably it's long term task...